sw/Core/Src/touch.h

```diff
@@ -551,33 +553,62 @@ u8 find_midi_note(u8 chan, u8 note) {
 		return fi;
 	return 255;
 }
-u8 find_midi_free_channel(void) {
-	// "dumb" implementation: just find the lowest unused string
-	for (u8 i = 0; i < 8; i++)
-		if (!(synthfingerdown_nogatelen_internal & (1 << i)))
-			return i;
-	// no unused string found
-	return 255;
-
-	// RJ: I'm really not quite sure what this code was doing exactly. Keeping it here for reference
+u8 find_free_midi_string(u8 midi_pitch) {
+	// RJ: I would like to map incoming midi in a way where the difference between the midi pitch
+	// and the average pitch of the string it lands on is as small as possible. This way we try to
+	// keep midi notes on strings that would otherwise play pitches (by hand or sequencer) roughly in
+	// the same range. When touch- or sequencer-priority cuts off a midi note, this should lead to the
+	// least-jarring results. Note-glides would also make sense this way as we try to keep pitches close
 	//
-	// u8 numfingerdown = 0;
-	// for (int attempt = 0; attempt < 16; ++attempt) {
-	// 	u8 ch = (midi_next_finger++) & 7;
-	// 	bool fingerdown = touch_synth_getlatest(ch)->pressure > 0 && (midi_pressure_override&(1<<ch))==0; // synth_dst_finger>0, and midi_pressure_override bit is not set
-	// 	if (fingerdown) {
-	// 		if (attempt < 8) {
-	// 			numfingerdown++;
-	// 			if (numfingerdown == 8)
-	// 				return 255; // all fingers using channels! NO ROOM! TOUGH
-	// 		}
-	// 		continue;
-	// 	}
-	// 	if (attempt >=8 || midi_channels[ch] == 255)
-	// 		return ch;
-	// }
-	// // give up
-	// return (midi_next_finger++) & 7;
+	// Reverse-engineering the central pitch of a string from the calculations in DoAudio() is a little too 
+	// complex for me to figure out. Is there a simpler way to get those? For now I'm implemeting a very
+	// simple way of mapping midi notes over the strings from low to high.
+
```

Comment:
i agree its super gnarly. my suggestion to avoid a big refactor would actually be for the DoAudio() routine to maintain a 'lowest note' and 'highest note' in each string, or even just a midpoint, and then have this code swoop in and compare to that?